libqtile/widget/base.py

```diff
@@ -311,12 +312,17 @@ def calculate_length(self):
 
     def timeout_add(self, seconds, method, method_args=()):
         """
-            This method calls either ``.call_later`` with given arguments.
+            This method calls ``.call_later`` with given arguments.
         """
-        self.future = self.qtile.call_later(
+        future = self.qtile.call_later(
             seconds, self._wrapper, method, *method_args
         )
 
+        # Remove completed timers from the list and add this timer
+        self._futures = [timer for timer in self._futures if timer._scheduled]
```

Comment:
Is there a way we can avoid using the private `_scheduled` property?

Reply:
Ha. Yeah, I didn't love that either. There are very few public properties. I don't think `cancelled` works.

Reply:
You could compare `timer.when()` with `self.qtile._eventloop.time()` and, if the timer has expired, remove it.

Reply:
@m-col any preference on this one?

Reply:
Why doesn't `cancelled` alone work?

I remember there being a way to get the return value of scheduled functions, though the docs for this aren't revealing this to me now. Perhaps it wasn't `TimerHandle` or `Handle` but another way to schedule things. I'll keep looking.

Reply:
`cancelled` is only `True` if the timer is cancelled, not if it has been run.

Reply:
Oh yeah, du doi. In that case yeah this works.